packages/firestore/exp/src/api/transaction.ts

```diff
@@ -103,7 +103,7 @@ export function runTransaction<T>(
   firestore._queue.enqueueAndForget(async () => {
     const datastore = await getDatastore(firestore);
     new TransactionRunner<T>(
-      new AsyncQueue(),
+      firestore._queue,
```

Comment:
Note: This is a bug in the firestore-exp code, which breaks a test. See how `firestoreClientTransaction` using the main AsyncQueue here.